tests/components/tessie/test_button.py

```diff
@@ -1,39 +1,47 @@
 """Test the Tessie button platform."""
 from unittest.mock import patch
 
-import pytest
+from syrupy import SnapshotAssertion
 
 from homeassistant.components.button import DOMAIN as BUTTON_DOMAIN, SERVICE_PRESS
-from homeassistant.const import ATTR_ENTITY_ID
+from homeassistant.const import ATTR_ENTITY_ID, Platform
 from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
 
 from .common import setup_platform
 
 
-@pytest.mark.parametrize(
-    ("entity_id", "func"),
-    [
+async def test_buttons(
+    hass: HomeAssistant, snapshot: SnapshotAssertion, entity_registry: er.EntityRegistry
+) -> None:
+    """Tests that the button entities are correct."""
+
+    entry = await setup_platform(hass, [Platform.BUTTON])
+
+    entity_entries = er.async_entries_for_config_entry(entity_registry, entry.entry_id)
+
+    assert entity_entries
+    for entity_entry in entity_entries:
+        assert entity_entry == snapshot(name=f"{entity_entry.entity_id}-entry")
+        assert hass.states.get(entity_entry.entity_id) == snapshot(
+            name=f"{entity_entry.entity_id}-state"
+        )
```

Comment:
You can extract this into a function.

Also, Robert had some improvements on this kind of test. checkout Airly, which has the improvements

Reply:
Done this.